File: src-tauri/src/global_state.rs
use tauri::command;
use tokio::sync::Mutex;

use snow_shot_global_state::CaptureState;

#[command]
pub async fn set_capture_state(
    capture_state: tauri::State<'_, Mutex<CaptureState>>,
    capturing: bool,
) -> Result<(), String> {
    let mut capture_state = capture_state.lock().await;
    capture_state.capturing = capturing;

    Ok(())
}

#[command]
pub async fn get_capture_state(
    capture_state: tauri::State<'_, Mutex<CaptureState>>,
) -> Result<CaptureState, String> {
    let capture_state = capture_state.lock().await;
    Ok(capture_state.clone())
}

File: src/commands/global_state.ts
import { invoke } from "@tauri-apps/api/core";
import type { CaptureState } from "@/types/commands/global_state";

export const setCaptureState = async (capturing: boolean) => {
	const result = await invoke<void>("set_capture_state", { capturing });
	return result;
};

export const getCaptureState = async () => {
	const result = await invoke<CaptureState>("get_capture_state");
	return result;
};
